### PR TITLE
feat(require-2fa): handle saml and require 2fa

### PR DESCRIPTION
**Pull request by maheskett**
- Disable require 2fa when saml is setup
- Restrict requiring 2fa once saml is setup
screen shot 2018-09-05 at 2 44 12 pm
screen shot 2018-09-05 at 12 21 54 am

### CODE REVIEW

File: src/sentry/auth/providers/saml2.py [hunk 1]
```diff
@@ -378,3 +387,20 @@ def build_auth(request, saml_config):
...
+        organization.update(
+            flags=models.F('flags').bitand(~Organization.flags.require_2fa)
+        )
+
+        create_audit_entry(
+            request=request,
+            organization=organization,
+            target_object=organization.id,
+            event=AuditLogEntryEvent.ORG_EDIT,
+            data={'require_2fa': u'to False'},
```
Comment (markstory) [inline src/sentry/auth/providers/saml2.py]: Should this include the why (saml was enabled) as well?

File: src/sentry/static/sentry/app/views/settings/organizationGeneralSettings/organizationSettingsForm.jsx [hunk 2]
```diff
@@ -42,14 +41,6 @@ const OrganizationSettingsForm = createReactClass({
             onSave(initialData, model.initialData);
           }
         }}
-        onSubmitError={error => {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/organizationGeneralSettings/organizationSettingsForm.jsx]: Are there no other failure conditions anymore? Should you keep the `Unable to save change` error?

**Review by markstory — COMMENTED** [review]
Looks good to me, but I know very little about our saml implementation.

File: src/sentry/auth/providers/saml2.py [hunk 3]
```diff
@@ -67,6 +77,7 @@ def dispatch(self, request, helper):
         # not have been configured yet, build the config first from the state
         if not provider.config:
             provider.config = provider.build_config(helper.fetch_state())
+            disable_2fa_required(request, helper.organization)
```
Comment (EvanPurkhiser) [inline src/sentry/auth/providers/saml2.py]: Could you update the comment above this if to read more along the lines of 'During the setup pipeline these two things need to happen'

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**